src/mount/writedata.cc

```diff
@@ -863,7 +942,8 @@ static void	write_data_lcnt_decrease_check_deleted(inodedata *id, Glock& lock, b
 	// As long as it is not freed, then we don't consider the inodedata deleted
 	isDeleted = false;
 	id->lcnt--;
-	if (id->lcnt == 0 && !id->inqueue && id->flushwaiting == 0 && id->writewaiting == 0) {
+	if (id->lcnt == 0 && id->chunksDataFirst == nullptr &&
```

Comment:
This could be another place to consider to avoid comparison directly with nullptr and put a constant or macro instead.

Reply:
This specific point was very much changed due to other suggestions.